src/charm.py

```diff
@@ -307,6 +313,62 @@ def check_pgb_running(self):
 
         return True
 
+    def get_hostname_by_unit(self, unit_name: str) -> str:
+        """Create a DNS name for a PgBouncer unit.
+
+        Args:
+            unit_name: the juju unit name, e.g. "postgre-sql/1".
+
+        Returns:
+            A string representing the hostname of the PostgreSQL unit.
+        """
+        unit_id = unit_name.split("/")[1]
+        return f"{self.app.name}-{unit_id}.{self.app.name}-endpoints"
+
+    def get_secret(self, scope: str, key: str) -> Optional[str]:
+        """Get secret from the secret storage."""
+        return self.peers.get_secret(scope, key)
+
+    def set_secret(self, scope: str, key: str, value: Optional[str]) -> None:
+        """Set secret from the secret storage."""
+        self.peers.set_secret(scope, key, value)
+
+    def push_tls_files_to_workload(self, container: Container = None) -> None:
+        """Uploads TLS files to the workload container."""
+        if container is None:
+            container = self.unit.get_container("pgbouncer")
+
+        key, ca, cert = self.tls.get_tls_files()
+        if key is not None:
+            container.push(
+                f"{PGB_DIR}/{TLS_KEY_FILE}",
+                key,
+                make_dirs=True,
+                permissions=0o400,
+                user=PG_USER,
+                group=PG_GROUP,
+            )
+        if ca is not None:
+            container.push(
+                f"{PGB_DIR}/{TLS_CA_FILE}",
+                ca,
+                make_dirs=True,
+                permissions=0o400,
+                user=PG_USER,
+                group=PG_GROUP,
+            )
+        if cert is not None:
+            container.push(
+                f"{PGB_DIR}/{TLS_CERT_FILE}",
+                cert,
+                make_dirs=True,
+                permissions=0o400,
+                user=PG_USER,
+                group=PG_GROUP,
+            )
+
+        # TODO update config
```

Comment:
TBD in [DPE-558](https://warthogs.atlassian.net/browse/DPE-558)

Reply:
`update_config()` is called by the tls lib when removing relation with `tls-certificates-operator`